### PR TITLE
28 notifications apis

### PR DESCRIPTION
**Pull request by W-Shih**
- feature: add notifications list, unread_count, and mark_all_as_read apis
- test: add tests for above apis
- doc: update pylint score, comments, and history

### CODE REVIEW

File: inbox/api/views.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+from rest_framework.decorators import action
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from inbox.api.serializers import NotificationSerializer, DefaultNotificationSerializer
+
+
+class NotificationViewSet(viewsets.GenericViewSet):
+    queryset = Notification.objects.all()
```
Comment (wen-wayne-shih) [inline inbox/api/views.py:28]: you can also override get_queryset(self) to only return Notification.objects.filter(recipient=self.request.user)

File: inbox/api/views.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    def list(self, request: Request):
+        queryset = self.get_queryset()
+        notifications = self.filter_queryset(queryset)\
+            .filter(recipient=request.user)\
+            .order_by('-timestamp')\
+            .prefetch_related('actor', 'target')
+        serializer = NotificationSerializer(notifications, many=True, context={'request': request})
+        return Response({
+            'notifications': serializer.data,
+        }, status=status.HTTP_200_OK)
```
Comment (wen-wayne-shih) [inline inbox/api/views.py:49]: you might also consider to use ListModelMixin

**Review by wen-wayne-shih — COMMENTED** [review]
Plz refer to per comments

File: inbox/api/views.py (hunk 1 above)
Comment (W-Shih) [reply inbox/api/views.py:28]: I will think about the other use cases and consider this suggestion.
If so, I will make the change with the next notification PR together.

File: inbox/api/views.py (hunk 2 above)
Comment (W-Shih) [reply inbox/api/views.py:49]: Not sure if ListModelMixin supports prefetch_related?

Comment (wen-wayne-shih) [reply inbox/api/views.py:49]: In this case, you could try to make 
filter(), order_by(), and prefetch_related() all in the get_queryset()

File: inbox/api/views.py (hunk 1 above)
Comment (wen-wayne-shih) [reply inbox/api/views.py:28]: If you do override get_queryset(), consider your use case, you could try to make
filter(), order_by(), and prefetch_related() all in the get_queryset()

**Review by wen-wayne-shih — APPROVED** [review]

**Pull request merged**